src/libraries/System.Private.CoreLib/src/System/SearchValues/Any1CharPackedIgnoreCaseSearchValues.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Buffers
+{
+    internal sealed class Any1CharPackedIgnoreCaseSearchValues : SearchValues<char>
+    {
+        private readonly char _lowerCase, _upperCase;
+
+        public Any1CharPackedIgnoreCaseSearchValues(char value)
+        {
+            Debug.Assert((value | 0x20) == value);
+
+            _lowerCase = value;
+            _upperCase = (char)(value & ~0x20);
```

Comment:
Technically these might not be lower and upper case, right?  I don't see a check being performed at the construction site to validate that these are ASCII letter, so really they're just any two characters that differ by this bit, which includes but isn't limited to ASCII letter cased pairings.

Reply:
That's right, I mentioned that in [one place](https://github.com/dotnet/runtime/blob/73332dd0f0da3e250a7d8c42a352c943a08dd6c5/src/libraries/System.Private.CoreLib/src/System/SearchValues/SearchValues.cs#L131), but it makes sense to call out here as well.